### PR TITLE
rewrote example to OO format

### PR DESCRIPTION
**Pull request by dmgt**
## PR Summary

One more PR from the SciPy sprint. Rewrote this example to use the OO format instead of the older API

## PR Checklist
- [ x] Code is PEP 8 compliant

### CODE REVIEW

**Comment by choldgraf** [comment]
Looks good to me - it seems like the circle error isn't related to this PR but @QuLogic do you have an intuition for that?

File: examples/pylab_examples/log_demo.py [hunk 1]
```diff
@@ -9,37 +9,41 @@
...
+# expressed as a fraction of the average axis height
+fig.subplots_adjust(hspace=0.5)
+
 # log y axis
-plt.subplot(221)
-plt.semilogy(t, np.exp(-t/5.0))
-plt.title('semilogy')
-plt.grid(True)
+ax1.semilogy(t, np.exp(-t/5.0))
+ax1.set(title='semilogy')
```
Comment (dopplershift) [inline examples/pylab_examples/log_demo.py]: Any particular reason for this instead of:
```python
ax1.set_title('semilogy')
```
?

Comment (dmgt) [inline examples/pylab_examples/log_demo.py]: No particular reason - in other pylab examples there are usually multiple keyword arguments being passed to ax.set but in this case only one (title). If way or the other is preferred, happy to update it.

**Comment by QuLogic** [comment]
I think the CircleCI failure is because this branch was opened before that PR was merged, so it's using the default build steps. A rebase or opening/closing might fix it.

**Comment by QuLogic** [comment]
No luck :( In any case, I don't think there's anything here that should break the docs, so it's probably not worth the effort to rebase unless you're making other changes.

**Comment by choldgraf** [comment]
ok cool - I'm gonna merge it in that case...will make a note to look over these examples in the next PR :-)

**Pull request merged**